File: CPP_Solutions/Knapsack/Zero_One_Knapsack/Zero_One_Knapsack_Recursive.cpp
#include <iostream>

using namespace std;

int knapsack(int wt[], int val[], int W, int n) {
    if(n == 0 || W == 0)
        return 0;
    else if(wt[n-1] <= W)
        return std::max(val[n-1] + knapsack(wt, val, W - wt[n-1], n-1),knapsack(wt, val, W, n-1));
    else //wt[n-1] > W
        return knapsack(wt, val, W, n-1);
}

int main()
{
    // cout<<"Hello World";
    int weight[] = { 10, 20, 30 };
    int val[] = { 60, 100, 120 };
    int W = 50;
    int n = sizeof(val) / sizeof(val[0]);
    cout << knapsack(weight, val, W, n);

    return 0;
}

File: Python_Solutions/Knapsack/Zero_One_Knapsack/Zero_One_Knapsack_Recursive.py
def knapSack( wt, val, W, n):
    if (n == 0 or W == 0):
        return 0
    elif (wt[n-1] <= W):
        return max(
            val[n-1] + knapSack( wt, val, W-wt[n-1], n-1),
            knapSack( wt, val, W, n-1))        
    else:
        return knapSack(W, wt, val, n-1)

val = [60, 100, 120]
wt = [10, 20, 30]
W = 50
n = len(val)
print(knapSack(wt, val, W, 3))